custom_components/tuya_local/devices/etop_fcu_thermostat.yaml

```diff
@@ -0,0 +1,141 @@
+name: ETOP-FCU(CF) thermostat
+products:
+  - id: psetc59fvtm5gexl
+    name: Jaga JRT-100TW
+primary_entity:
+  entity: climate
+  name: Thermostat
+  dps:
+    ############################################################
+    # Setpoint and current temperature
+    ############################################################
+    # Using DPs for current temperature and setpoint in °C
+    # if you want to control in °F instead, comment these DPs
+    # and uncomment the DPs for °F below
+    - id: 2
+      name: temperature
+      type: integer
+      unit: C
+      range:
+        min: 50
+        max: 350
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 3
+      name: current_temperature
+      type: integer
+      unit: C
+      mapping:
+        - scale: 10
+          step: 5
+    # Available DPs for current temperature and setpoint in °F
+    # if you want to control in °F, uncomment these DPs
+    # and comment the DP for °C above
+    # - id: 20
+    #   name: temperature
+    #   hidden: true
+    #   type: integer
+    #   unit: F
+    #   range:
+    #     min: 41
+    #     max: 95
+    #   mapping:
+    #     - scale: 1
+    #       step: 1
+    # - id: 21
+    #   name: current_temperature
+    #   hidden: true
+    #   type: integer
+    #   unit: F
+    #   range:
+    #     min: 32
+    #     max: 122
+    #   mapping:
+    #     - scale: 1
+    #       step: 1
+    ############################################################
+    # Thermostat modes
+    #
+    # Depending on your actual heating/cooling installation
+    # you can hide the unnecessary entries
+    - id: 4
+      name: hvac_mode
+      type: string
+      mapping:
+        - dps_val: fan
+          value: fan_only
+        - dps_val: heat
+          value: heat
+        - dps_val: cold
+          value: cool
+        - dps_val: auto
+          value: auto
+    - id: 5
+      name: fan_mode
+      type: string
+      mapping:
+        - dps_val: low
+          value: low
+        - dps_val: middle
+          value: medium
+        - dps_val: high
+          value: high
+        - dps_val: auto
+          value: auto
+secondary_entities:
+  # Overall On/Off switch
+  - entity: switch
```

Comment:
I tried the following, which is exactly what I've seen in other config files, but it fails to load completely, and I have no clue why:


```
    - id: 1
      name: hvac_mode
      type: boolean
      mapping:
      	- dps_value: false
          value: "off"
        - dps_value: true
          constraint: preset_mode
          conditions:
            - dps_val: fan
              value: fan_only
            - dps_val: heat
              value: heat
            - dps_val: cold
              value: cool
            - dps_val: auto
              value: heat_cool
    - id: 4
      name: preset_mode
      type: string 
      hidden: true
```

Reply:
You have put dps_value where it should be dps_val

I would choose a name other than preset_mode though - work_mode or just mode, as you do not need hvac_mode duplicated as preset_mode (some configs use that because they have additional modes that hvac_mode does not cover).

Reply:
Fixed the wrong parameter name (sorry for missing that) and renamed the 2nd dps to work_mode.
Everything is working as expected now!